src/java.base/share/classes/java/lang/reflect/Method.java

```diff
@@ -381,7 +383,13 @@ public boolean equals(Object obj) {
      * method's declaring class name and the method's name.
      */
     public int hashCode() {
-        return getDeclaringClass().getName().hashCode() ^ getName().hashCode();
+        int hc = hash;
+
+        if (hc == 0) {
```

Comment:
Should we add an extra field `hashIsZero` like in `String` to avoid repeated computation if the hash is 0?